tests/fixture/testnet/local/network.go

```diff
@@ -0,0 +1,644 @@
+	flags[cfg.NetworkNameKey] = fmt.Sprintf("%d", ln.Genesis.NetworkID)
+
+	// Ensure keys are added if necessary
+	if err := node.EnsureKeys(); err != nil {
+		return err
+	}
+
+	// Ensure the node's data dir is configured
+	dataDir := node.GetDataDir()
+	if len(dataDir) == 0 {
+		// NodeID will have been set by EnsureKeys
+		dataDir = filepath.Join(ln.Dir, node.NodeID.String())
+		flags[cfg.DataDirKey] = dataDir
+	}
+
+	return nil
+}
+
+// Starts a network for the first time
+func (ln *LocalNetwork) Start(w io.Writer) error {
+	if len(ln.Dir) == 0 {
+		return errLocalNetworkDirNotSet
+	}
+
+	// Ensure configuration on disk is current
+	if err := ln.WriteAll(); err != nil {
+		return err
+	}
+
+	// Accumulate bootstrap nodes such that each subsequently started
+	// node bootstraps from the nodes previously started.
+	//
+	// e.g.
+	// 1st node: no bootstrap nodes
+	// 2nd node: 1st node
+	// 3rd node: 1st and 2nd nodes
+	// ...
+	//
+	bootstrapIDs := []string{}
+	bootstrapIPs := []string{}
+
+	// Configure networking and start each node
+	for _, node := range ln.Nodes {
+		// Update network configuration
+		node.SetNetworkingConfigDefaults(0, 0, bootstrapIDs, bootstrapIPs)
+
+		// Write configuration to disk in preparation for node start
+		if err := node.WriteConfig(); err != nil {
+			return err
+		}
+
+		// Start waits for the process context to be written which
+		// indicates that the node will be accepting connections on
+		// its staking port. The network will start faster with this
+		// synchronization due to the avoidance of exponential backoff
+		// if a node tries to connect to a beacon that is not ready.
+		if err := node.Start(w, ln.ExecPath); err != nil {
+			return err
+		}
+
+		// Collect bootstrap nodes for subsequently started nodes to use
+		bootstrapIDs = append(bootstrapIDs, node.NodeID.String())
+		bootstrapIPs = append(bootstrapIPs, node.StakingAddress)
+	}
+
+	return nil
+}
+
+// Wait until all nodes in the network are healthy.
+func (ln *LocalNetwork) WaitForHealthy(ctx context.Context, w io.Writer) error {
+	ticker := time.NewTicker(networkHealthCheckInterval)
+	defer ticker.Stop()
+
+	healthyNodes := set.NewSet[ids.NodeID](len(ln.Nodes))
+	for healthyNodes.Len() < len(ln.Nodes) {
+		for _, node := range ln.Nodes {
+			if healthyNodes.Contains(node.NodeID) {
+				continue
+			}
+
+			healthy, err := node.IsHealthy(ctx)
+			if err != nil && !errors.Is(err, errProcessNotRunning) {
+				return err
+			}
+			if !healthy {
+				continue
+			}
+
+			healthyNodes.Add(node.NodeID)
+			if _, err := fmt.Fprintf(w, "%s is healthy @ %s\n", node.NodeID, node.URI); err != nil {
+				return err
+			}
+		}
+
+		select {
+		case <-ctx.Done():
+			return errNotHealthyBeforeTimeout
+		case <-ticker.C:
+		}
+	}
+	return nil
+}
+
+// Retrieve API URIs for all nodes in the network. Assumes nodes have
+// been loaded.
+func (ln *LocalNetwork) GetURIs() []string {
+	uris := []string{}
```

Comment:
Not sure we really care to pre-alloc stuff like this in the testing harness... But we can also pre-alloc here.